babel.config.cjs

```diff
@@ -31,5 +31,19 @@ module.exports = {
         "@babel/plugin-transform-object-rest-spread",
         "@babel/plugin-syntax-dynamic-import",
         "@babel/plugin-transform-runtime",
+        [
+            "search-and-replace",
+            {
+                rules:
+                    process.env.NODE_ENV === "test"
+                        ? []
+                        : [
+                              {
+                                  search: 'import("./rust-crypto")',
+                                  replace: 'import("./rust-crypto/index.js")',
```

Comment:
I'm a bit confused to be honest. So this is necessary because in nodejs environments we need one import style and elsewhere we need another, and there isn't one that works everywhere? Surely this is only evaluated at js-sdk build time so we'd just end up with one or the other depending on how the js-sdk gets babel-ed?

Reply:
Node ESM requires that the full path be set:
https://nodejs.org/api/esm.html#esm_mandatory_file_extensions

Jest tends not to play with ESM that well, so AFAIK your tests are needing `.ts` extensions which of course will need to be re-written when compiling to JS. The babel re-write of imports don't work on dynamic imports (yet) so that is why the extension was omitted.

Reply:
This does look plausible to me, though again I think it needs code comments to explain it to future readers.

I think basically it's the same thing we ended up with in https://github.com/matrix-org/matrix-js-sdk/pull/4377: basically, whenever we're using the transpiled javascript in `lib`, we want to be using `*.js` imports: it's the only thing that keeps node.js happy. tsc is also happy with `.js` imports if it sees them: it will magically use a `.ts` instead of `.js`. The fly in the ointment is jest, which takes the original typescript source and feeds it through babel on the fly: it does not have the `.js`->`.ts` intelligence, so doesn't find the right typescript files.

@saul-jb: if you could try to explain some of this in a code comment, that would be very helpful.